src/MPCC.cpp

```diff
@@ -468,11 +479,19 @@ int pcc_matrix(int m, int n, int p,
     //Compute and assemble composite terms
 
     //SASB=SA*SB
-    VMUL(m*p,SA,SB,SASB);
-    //N*SASB
+    if(!sameAB)
+      VMUL(m*p,SA,SB,SASB);
+    else {
+      #pragma omp parallel for private(i, j)
```

Comment:
This code is duplicated two times, why not make it a function? 
Also the code you provide now doesn't benefit from MKL vectorization like the VMUL function

Reply:
There is no difference in performance in the above code as compared to VMUL. I think that the OMP multithreading is equivalent to VMUL implementation. Testing on penguin2 gave nearly equal execution time.